hack/install-wrapper.sh

```diff
@@ -0,0 +1,250 @@
+AWS_METADATA_QUERY_ID="http://169.254.169.254/latest/meta-data/instance-id"
+AZURE_METADATA_QUERY_ID="http://169.254.169.254/metadata/instance/compute/resourceId?api-version=2021-02-01&format=text"
+AZURE_METADATA_QUERY_NAME="http://169.254.169.254/metadata/instance/compute/name?api-version=2021-02-01&format=text"
+
+# Platform specific
+OS=""
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information"
+}
+
+_usage="Usage: $0 [arguments]
+
+Downloads install script from Antrea and setup antrea-agent on AWS or Azure VM.
+
+[arguments]
+        --ns <Namespace>                                Namespace to be used by the antrea-agent.
+        --kubeconfig <KubeconfigSavePath>               Path of the kubeconfig to access K8s API Server.
+        --antrea-kubeconfig <AntreaKubeconfigSavePath>  Path of the kubeconfig to access Antrea API Server.
+        --antrea-version <Version>                      Antrea version to be used.
+        --bin <AntreaAgentSavePath>                     Path of the antrea-agent binary(Temporary).
+        --help, -h                                      Print this message and exit."
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --ns)
+    NAMESPACE="$2"
+    shift 2
+    ;;
+    --kubeconfig)
+    KUBECONFIG="$2"
+    shift 2
+    ;;
+    --antrea-kubeconfig)
+    ANTREA_KUBECONFIG="$2"
+    shift 2
+    ;;
+    --antrea-version)
+    ANTREA_VERSION="$2"
+    shift 2
+    ;;
+    --bin)
+    AGENT_BIN="$2"
+    shift 2
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *)    # unknown option
+    echoerr "Unknown option $1"
+    print_help
+    exit 1
+    ;;
+esac
+done
+
+if [ -z "$ANTREA_VERSION" ] || [ -z "$NAMESPACE" ] || [ -z "$KUBECONFIG" ] || [ -z "$ANTREA_KUBECONFIG" ]; then
+  echo "Required fields are not set"
+  print_usage
+  exit 1
+fi
+
+# Generic function to execute command with timeout.
+function execute_command() {
+    cmd="$1"
+    timeout="$2"
+    echo "Executing $cmd with timeout $timeout at $(date)"
+    timeout --preserve-status --foreground $timeout $cmd
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        echoerr "$cmd failed with status $ret at $(date)"
+        echoerr "Exiting!! please check internet connectivity and re-run install script"
+        exit 2
+    fi
+}
+
+function get_cloud_platform() {
+    cat /sys/class/dmi/id/bios_version | grep -iq "amazon"
+    if [ $? -eq 0 ]; then
+        echo -n "$AWS"
+        return
+    fi
+    cat /sys/class/dmi/id/bios_vendor | grep -iq "amazon"
+    if [ $? -eq 0 ]; then
+        echo -n "$AWS"
+        return
+    fi
+    cat /sys/class/dmi/id/sys_vendor | grep -iq "microsoft"
+    if [ $? -eq 0 ]; then
+        echo -n "$AZURE"
+        return
+    fi
+}
+
+function check_os_prerequisites() {
+    CLOUD=$(get_cloud_platform)
+    if [ -z "$CLOUD" ]; then
+        echoerr "Unknown cloud platform, supported cloud platforms: AWS and Azure"
```

Comment:
nit: Unknown cloud platform. Only AWS and Azure Clouds are supported.

Reply:
Done